File: week-04/day-3/anagram.py
def is_anagram(str1, str2):
    str1 = str.lower(str1)
    str2 = str.lower(str2)
    
    list1 = ''.join(sorted(str1))
    list2 = ''.join(sorted(str2))

    if list1.replace(" ", "") == list2.replace(" ", ""):
        return True
    else:
        return False

File: week-09/day-3/anagram.js
/*let anagram = (s1, s2) =>
  s1.split('').sort().join('') === s2.split('').sort().join('')*/


let anagram = function(string1, string2) {
  if (typeof string1 !== 'string') {
    throw new Error ('string1 is not a string!');
  } else if (typeof string2 !== 'string') {
    throw new Error ('string2 is not a string!');
  }
  
  let str1 = string1.toLowerCase().replace(/[^a-z\d]/g, '').split('').sort('').join('');
  let str2 = string2.toLowerCase().replace(/[^a-z\d]/g, '').split('').sort('').join('');
  
  if (str1.length !== str2.length) {
    throw new Error ('The two string are not an anagram!');
  } else if (str1 === str2) {
    return true;
  }
}

module.exports = anagram;

//console.log(anagram(123, 231));
